File: frontend/src/types/report.ts
export type ReportStatus = "pending" | "accepted" | "needs_revision";

export const REPORT_STATUS_LABELS: Record<ReportStatus, string> = {
  pending: "На проверке",
  accepted: "Принят",
  needs_revision: "Нужно доработать",
};

export interface Report {
  trainingId: string;
  submittedBy: string;
  textReport: string;
  photoFileId: string | null;
  videoFileId: string | null;
  status: ReportStatus;
  coachComment: string | null;
  reviewedBy: string | null;
  reviewedAt: string | null;
}

export interface ReportDto {
  training_id: string;
  submitted_by: string;
  text_report: string;
  photo_file_id: string | null;
  video_file_id: string | null;
  status: ReportStatus;
  coach_comment: string | null;
  reviewed_by: string | null;
  reviewed_at: string | null;
}

export function mapReportDto(dto: ReportDto): Report {
  return {
    trainingId: dto.training_id,
    submittedBy: dto.submitted_by,
    textReport: dto.text_report,
    photoFileId: dto.photo_file_id,
    videoFileId: dto.video_file_id,
    status: dto.status,
    coachComment: dto.coach_comment,
    reviewedBy: dto.reviewed_by,
    reviewedAt: dto.reviewed_at,
  };
}

File: backend/app/schemas/report.py
from __future__ import annotations

from datetime import datetime
from typing import Literal
from uuid import UUID

from pydantic import BaseModel, Field

ReportStatus = Literal["pending", "accepted", "needs_revision"]
ReviewDecision = Literal["accepted", "needs_revision"]


class ReportSubmitIn(BaseModel):
    text_report: str = Field(min_length=1, max_length=2000)


class ReportReviewIn(BaseModel):
    decision: ReviewDecision
    coach_comment: str | None = Field(default=None, max_length=1000)


class ReportOut(BaseModel):
    training_id: UUID
    submitted_by: UUID
    text_report: str
    photo_file_id: UUID | None
    video_file_id: UUID | None
    status: ReportStatus
    coach_comment: str | None
    reviewed_by: UUID | None
    reviewed_at: datetime | None
    created_at: datetime
    updated_at: datetime
